src/App.tsx

```diff
@@ -15,36 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum ESortType {
+  ALPHABETICALLY = 'alphabetically',
+  LENGTH = 'by length',
+  DEFAULT = '',
+}
+
+interface IOptions {
+  sortBy: ESortType;
+  isReversed: boolean;
+}
+
+const getGoods = (goods: string[], { sortBy, isReversed }: IOptions) => {
+  let visibleGoods = [...goods];
+
+  if (isReversed) {
+    visibleGoods = visibleGoods.toReversed();
+  }
+
+  if (sortBy) {
+    switch (sortBy) {
+      case ESortType.ALPHABETICALLY:
+        visibleGoods = visibleGoods.toSorted();
+        if (isReversed) {
+          visibleGoods = visibleGoods.toReversed();
+        }
+
+        break;
+      case ESortType.LENGTH:
+        if (isReversed) {
+          visibleGoods = visibleGoods.toSorted((a, b) => b.length - a.length);
+        } else {
+          visibleGoods = visibleGoods.toSorted((a, b) => a.length - b.length);
+        }
```

Comment:
what if we move reverse logic to the end (after switch) and execute it only there? Will it work as expected? I mean if we already sorted by length - having reversed it we'll get the expected order anyway, no need to sort in reverse direction. But I might be missing sth